prow/cmd/checkconfig/main.go

```diff
@@ -194,6 +196,8 @@ func validateTideRequirements(configAgent config.Agent, pluginAgent plugins.Plug
 		{plugin: releasenote.PluginName, label: releasenote.ReleaseNoteLabelNeeded, matcher: forbids},
 		{plugin: cherrypickunapproved.PluginName, label: cherrypickunapproved.CpUnapprovedLabel, matcher: forbids},
 		{plugin: blockade.PluginName, label: blockade.BlockedPathsLabel, matcher: forbids},
+		{plugin: trigger.PluginName, label: trigger.NeedsOkToTest, matcher: forbids},
```

Comment:
I'm wondering if rules like this one are going to be annoying to resolve. If we have a big tide query covering a lot of repos and it forbids `needs-ok-to-test` we don't necessarily have trigger enabled in all of those repos, it is just convenient to forbid the label everywhere because the query is otherwise appropriate.
These checks are certainly still useful in some cases so it would be nice if we had a mechanism to skip individual warnings instead of skipping entire classes of warnings with the `--warnings` flag.

Reply:
We turned on `trigger` for everything and it has not had ill effects AFAICT but yes if you feel like you will not do that, then we should not include this.

Reply:
Or call it out as a specific check that can be turned off

Reply:
Either of those options SGTM.
Including a blocker label in a query that doesn't require it in order to condense similar queries will probably be somewhat common. However, it would be great to be able to catch the case in which a blocker label is intentionally added to a query, but the plugin is accidentally left disabled.

If this logic existed as a plugin like the `verify-config` plugin we could replace failing the presubmit job with adding a `do-not-merge/*` label. By posting comments/reviews containing warnings, we could bring low confidence warnings to the PR authors attention without blocking the PR from merging. Reacting to a PR instead of running as a normal presubmit job would also make it easy to only consider the changes the PR author is making and ignore preexisting problems.

An exemption comment or flag like one that a linter would use to ignore errors could also work, but I'm not sure how that would be implemented and it wouldn't be user friendly like a bot comment.